recipes/glib/all/conanfile.py

```diff
@@ -89,90 +89,104 @@ def requirements(self):
             # for Linux, gettext is provided by libc
             self.requires("libgettext/0.21")
 
-        if tools.is_apple_os(self.settings.os):
+        if is_apple_os(self):
             self.requires("libiconv/1.17")
 
     def validate(self):
-        if hasattr(self, 'settings_build') and cross_building(self, skip_x64_x86=True):
-            raise ConanInvalidConfiguration("Cross-building not implemented")
-        if scm.Version(self.version) >= "2.69.0" and not self.options.with_pcre:
+        if Version(self.version) >= "2.69.0" and not self.options.with_pcre:
             raise ConanInvalidConfiguration("option glib:with_pcre must be True for glib >= 2.69.0")
-        if self.settings.os == "Windows" and not self.options.shared and scm.Version(self.version) < "2.71.1":
+        if self.settings.os == "Windows" and not self.options.shared and Version(self.version) < "2.71.1":
             raise ConanInvalidConfiguration(
                 "glib < 2.71.1 can not be built as static library on Windows. "
                 "see https://gitlab.gnome.org/GNOME/glib/-/issues/692"
             )
-        if scm.Version(self.version) < "2.67.0" and not is_msvc(self) and not self.options.with_elf:
+        if Version(self.version) < "2.67.0" and not is_msvc(self) and not self.options.with_elf:
             raise ConanInvalidConfiguration("libelf dependency can't be disabled in glib < 2.67.0")
 
     def build_requirements(self):
         self.tool_requires("meson/0.63.2")
         self.tool_requires("pkgconf/1.9.3")
 
     def source(self):
-        files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    @functools.lru_cache(1)
-    def _configure_meson(self):
-        meson = Meson(self)
-        defs = {}
-        if tools.is_apple_os(self.settings.os):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
```

Comment:
No `basic_layout()`?

Reply:
we were discussing if `meson_layout` is needed here https://github.com/conan-io/conan/issues/9713
apparently, it's not needed